std/pubkey.go

```diff
@@ -21,6 +21,8 @@ var _ types.PublicKeyCodec = DefaultPublicKeyCodec{}
 // Decode implements the PublicKeyCodec.Decode method
 func (cdc DefaultPublicKeyCodec) Decode(key *types.PublicKey) (crypto.PubKey, error) {
 	switch key := key.Sum.(type) {
+	case nil:
+		return crypto.PubKey(nil), nil
```

Comment:
can't we just do this before the `switch`:

```go
if key == nil {
  return nil
}
```

I don't think `nil` should be a switch `case`